cpp/unreal_plugins/UrdfBot/Source/UrdfBot/UrdfJointComponent.cpp

```diff
@@ -0,0 +1,106 @@
+            ConstraintInstance.SetOrientationDriveTwistAndSwing(true, false);
+            ConstraintInstance.SetAngularVelocityDriveTwistAndSwing(true, false);
+            ConstraintInstance.SetAngularDriveParams(spring, damping, force_limit);
+            break;
+        }
+    case UrdfJointType::Continuous:
+        {
+            float spring = FMath::RadiansToDegrees(joint_desc->damping_) * m_to_cm * 10.0f;
+            float damping = FMath::RadiansToDegrees(joint_desc->damping_) * m_to_cm;
+            float force_limit = FMath::RadiansToDegrees(joint_desc->effort_) * m_to_cm * 10.0f;
+
+            // use twist degree for single angular freedom with best PhysX optimization
+            ConstraintInstance.SetAngularTwistLimit(EAngularConstraintMotion::ACM_Free, 0.0f);
+            ConstraintInstance.SetOrientationDriveTwistAndSwing(true, false);
+            ConstraintInstance.SetAngularVelocityDriveTwistAndSwing(true, false);
+            ConstraintInstance.SetAngularDriveParams(spring, damping, force_limit);
+            break;
+        }
+    case UrdfJointType::Prismatic:
+        {
+            float linear_limit_size = (joint_desc->upper_ - joint_desc->lower_) * m_to_cm * m_to_cm * 0.5f;
+            float spring = joint_desc->damping_ * m_to_cm * m_to_cm * 10.0f;
+            float damping = joint_desc->damping_ * m_to_cm * m_to_cm;
+            float force_limit = joint_desc->effort_ * m_to_cm * m_to_cm;
+
+            ConstraintInstance.SetLinearXLimit(ELinearConstraintMotion::LCM_Limited, linear_limit_size);
+            ConstraintInstance.SetLinearDriveParams(spring, damping, force_limit);
+            ConstraintInstance.SetLinearPositionDrive(true, false, false);
+            break;
+        }
+    case UrdfJointType::Fixed:
+        break;
+    case UrdfJointType::Floating:
+        {
+            ConstraintInstance.SetLinearXLimit(ELinearConstraintMotion::LCM_Free, 0.0f);
+            ConstraintInstance.SetLinearYLimit(ELinearConstraintMotion::LCM_Free, 0.0f);
+            ConstraintInstance.SetLinearZLimit(ELinearConstraintMotion::LCM_Free, 0.0f);
+            ConstraintInstance.SetAngularTwistLimit(EAngularConstraintMotion::ACM_Free, 0.0f);
+            ConstraintInstance.SetAngularSwing1Limit(EAngularConstraintMotion::ACM_Free, 0.0f);
+            ConstraintInstance.SetAngularSwing2Limit(EAngularConstraintMotion::ACM_Free, 0.0f);
+            break;
+        }
+    case UrdfJointType::Planar:
+        {
+            float linear_limit_size = (joint_desc->upper_ - joint_desc->lower_) * m_to_cm * m_to_cm * 0.5f;
+
+            ConstraintInstance.SetLinearXLimit(ELinearConstraintMotion::LCM_Limited, linear_limit_size);
+            ConstraintInstance.SetLinearYLimit(ELinearConstraintMotion::LCM_Limited, linear_limit_size);
+            break;
+        }
+    default:
+        ASSERT(false);
+        break;
+    }
+
+    SetDisableCollision(true);
```

Comment:
Does this do the same thing as `ConstraintInstance.SetDisableCollision(true)`, i.e., are we doing the same thing twice?

Reply:
Yes. removed one of them